content/livesync/models/index.textile

```diff
@@ -0,0 +1,203 @@
+
+* @Subscribe@ for the channels you intend to subscribe to.
+* @History@ if you intend to "sync from historical messages":/livesync/models/models#history.
+
+h2(#install). Install
+
+The Models SDK requires a realtime client created using the "Ably JavaScript SDK":/getting-started/setup to interact with the Ably service.
+
+Install the Ably JavaScript SDK and the Models SDK from NPM:
+
+```[sh]
+npm install ably @ably-labs/models
+```
+
+Import the SDKs into your project:
+
+```[javascript]
+import ModelsClient from '@ably-labs/models';
+import { Realtime } from 'ably/promises';
+```
+
+Instantiate a realtime client using the Ably JavaScript SDK and pass the generated client into the Models constructor:
+
+```[javascript]
+const client = new Realtime.Promise({ key: '{{API_KEY}}' });
+const modelsClient = new ModelsClient({ client });
+```
+
+h3(#options). ClientOptions
+
+In addition to the underlying Ably realtime client, you can provide a number of other "@ClientOptions@":/api/realtime-sdk/types#client-options to configure the behavior of the Models SDK:
+
+- @syncOptions@ := is used to configure how the model state is synchronised via the Sync function.
+- @historyPageSize@ := is the limit used when "querying for paginated history":/livesync/models/models#history used to subscribe to changes from the correct point in the channel.
+- @messageRetentionPeriod@ := is the message retention period configured on the Ably channel. This is used to determine whether the model state can be brought up to date from "message history":/livesync/models/models#history rather than via a re-sync.
+- @retryStrategy@ := defines a retry strategy to use if calling the "sync function":/livesync/models/models#sync throws an error.
+- @eventBufferOptions@ := used to configure the in-memory sliding-window buffer used for reordering and deduplication.
+- @optimisticEventOptions@ := is used to configure how "optimistic events":/livesync/models/models#optimistic are applied.
+- @logLevel@ := configures the log level used to control the verbosity of log output. One of @fatal@, @error@, @warn@, @info@, @debug@, or @trace@.
+
+The following is an example of setting @ClientOptions@ when instantiating the Models SDK:
+
+```[javascript]
+const modelsClient = new ModelsClient({
+ client,
+ logLevel,
+ syncOptions: {
+  historyPageSize,
+  messageRetentionPeriod,
+  retryStrategy,
+ },
+ eventBufferOptions: {
+  bufferMs,
+  eventOrderer,
+ },
+ optimisticEventOptions: {
+  timeout,
+ },
+});
+```
+
+h2(#quickstart). Quickstart
+
+A model is a single instance of a live, observable data model backed by your database. In this guide, we will create a simple model that tracks a list of comments on a post.
+
+The state of our model will have the following type:
+
+```[javascript]
+type Post = {
+  id: number;
+  text: string;
+  comments: string[];
+};
+```
```

Comment:
These lines should appear under "Create a sync function" like:

> Create a simple sync function which loads the post and its comments.
> The state of our model will...
> The Models SDK will infer the type of the model state from the type of the data payload returned by the sync function.

Reply:
I'm not sure what you mean here sorry @mschristensen

Reply:
The type of the model is not relevant until we talk about the sync function, as the type of the model is inferred from the value of the `data` field returned by the sync function. So, I'm just suggesting that we move these lines L87-95 to appear under the "Create a sync function" heading on L115. My recommendation is that the new text under the "Create a sync function" heading is as described in my comment above. :)

Reply:
Fixed in this [commit](https://github.com/ably/docs/pull/2082/commits/c112ade0468d905e588f441cff4a895bd9b30ace)